internal/expr/expr_test.go

```diff
@@ -334,6 +338,49 @@ AND z = @sqlair_0 -- The line with $Person.id on it
 	inputArgs:      []any{HardMaths{X: 1, Y: 2}},
 	expectedParams: []any{1, 2},
 	expectedSQL:    "INSERT INTO arr VALUES (ARRAY[[1,2],[@sqlair_0,4]], ARRAY[[5,6],[@sqlair_1,8]]);",
+}, {
+	summary:        "single slice",
+	query:          "SELECT name FROM person WHERE id IN ($S[:])",
+	expectedParsed: "[Bypass[SELECT name FROM person WHERE id IN (] Input[S[:]] Bypass[)]]",
+	typeSamples:    []any{sqlair.S{}},
+	inputArgs:      []any{sqlair.S{1, 2, 3}},
+	expectedParams: []any{1, 2, 3},
+	expectedSQL:    "SELECT name FROM person WHERE id IN (@sqlair_0, @sqlair_1, @sqlair_2)",
+}, {
+	summary:        "many slices",
+	query:          "SELECT * AS &Person.* FROM person WHERE id IN ($Person.id, $S[:], $Manager.id, $IntSlice[:], $StringSlice[:])",
+	expectedParsed: "[Bypass[SELECT ] Output[[*] [Person.*]] Bypass[ FROM person WHERE id IN (] Input[Person.id] Bypass[, ] Input[S[:]] Bypass[, ] Input[Manager.id] Bypass[, ] Input[IntSlice[:]] Bypass[, ] Input[StringSlice[:]] Bypass[)]]",
+	typeSamples:    []any{sqlair.S{}, Person{}, Manager{}, IntSlice{}, StringSlice{}},
+	inputArgs:      []any{sqlair.S{2, 3, 4}, Person{ID: 1}, Manager{ID: 5}, IntSlice{6, 7, 8}, StringSlice{"9", "10", "11"}},
+	expectedParams: []any{1, 2, 3, 4, 5, 6, 7, 8, "9", "10", "11"},
+	expectedSQL:    "SELECT address_id AS _sqlair_0, id AS _sqlair_1, name AS _sqlair_2 FROM person WHERE id IN (@sqlair_0, @sqlair_1, @sqlair_2, @sqlair_3, @sqlair_4, @sqlair_5, @sqlair_6, @sqlair_7, @sqlair_8, @sqlair_9, @sqlair_10)",
+}, {
+	summary:        "slices and other expressions in IN statement",
+	query:          `SELECT name FROM person WHERE id IN ($S[:], func(1,2), "one", $IntSlice[:])`,
+	expectedParsed: `[Bypass[SELECT name FROM person WHERE id IN (] Input[S[:]] Bypass[, func(1,2), "one", ] Input[IntSlice[:]] Bypass[)]]`,
+	typeSamples:    []any{sqlair.S{}, IntSlice{}},
+	inputArgs:      []any{sqlair.S{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, IntSlice{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}},
+	expectedParams: []any{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+	expectedSQL:    `SELECT name FROM person WHERE id IN (@sqlair_0, @sqlair_1, @sqlair_2, @sqlair_3, @sqlair_4, @sqlair_5, @sqlair_6, @sqlair_7, @sqlair_8, @sqlair_9, func(1,2), "one", @sqlair_10, @sqlair_11, @sqlair_12, @sqlair_13, @sqlair_14, @sqlair_15, @sqlair_16, @sqlair_17, @sqlair_18, @sqlair_19)`,
+}, {
+	summary:        "slice of mixed types",
+	query:          "SELECT name FROM person WHERE id IN ($S[:])",
+	expectedParsed: "[Bypass[SELECT name FROM person WHERE id IN (] Input[S[:]] Bypass[)]]",
+	typeSamples:    []any{sqlair.S{}},
+	inputArgs:      []any{sqlair.S{1, "two", 3.0}},
+	expectedParams: []any{1, "two", 3.0},
+	expectedSQL:    "SELECT name FROM person WHERE id IN (@sqlair_0, @sqlair_1, @sqlair_2)",
+}, {
+	// No error is throw for an empty slice because we do not know where in the
```

Comment:
```suggestion
	// No error is thrown for an empty slice because we do not know where in the
```

The wording of "we do not know where in the query and how the slice input ..." is a little bit confusing. What about iterating over something like:
```
// No error is throw for an empty slice because we do not want to force the user to handle it as a different case, it is their choice.
```

Reply:
Sounds good, how about:
```
// No error is throw for when the user passes an empty slice because we do
// not want to limit the use of slices to only cases we have foreseen.
```

Reply:
That works.